calour/ms1_experiment.py

```diff
@@ -91,9 +92,107 @@ def heatmap(self, *args, **kwargs):
         '''
         if 'norm' not in kwargs:
             kwargs['norm'] = mpl.colors.LogNorm()
+        if 'mz_rt' in self.feature_metadata.columns:
+            if 'yticklabel_len' not in kwargs:
+                kwargs['yticklabel_len'] = None
+            if 'feature_field' not in kwargs:
+                kwargs['feature_field'] = 'mz_rt'
+            if 'yticklabel_kwargs' not in kwargs:
+                kwargs['yticklabel_kwargs'] = {'size': 6, 'rotation': 0}
         super().heatmap(*args, **kwargs)
 
     def __repr__(self):
         '''Return a string representation of this object.'''
         return 'MS1Experiment %s with %d samples, %d features' % (
             self.description, self.data.shape[0], self.data.shape[1])
+
+    def filter_mz(self, mz, tolerance=0.001, inplace=False, negate=False):
+        '''Filter metabolites based on m/z
+
+        Keep (or remove if negate=True) metabolites that have an m/z close (up to tolerance)
+        to the requested mz (or list of mz).
+
+        Parameters
+        ----------
+        mz: float or list of float
+            the M/Z to filter
+        tolerance: float, optional
+            the M/Z tolerance. filter metabolites with abs(metabolite_mz - mz) <= tolerance
+        inplace: bool, optional
+            True to replace current experiment, False to create new experiment with results
+        negate: bool, optional
+            If False, keep only metabolites matching mz
+            If True, remove metabolites matching mz
+
+        Returns
+        -------
+        calour.MS1Experiment
+            features filtered based on mz
+        '''
+        if 'MZ' not in self.feature_metadata.columns:
+            raise ValueError('The Experiment does not contain the column "MZ". cannot filter by mz')
+        mz = _to_list(mz)
+        keep = set()
+        for cmz in mz:
```

Comment:
better to use a boolean mask? like `select` in filtering.py

Reply:
changed to filter_mz_rt() to enable filtering based on mz / rt/ or both